packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -13,12 +13,21 @@ const PACKAGE_FILENAME = "package.json";
 const BABELIGNORE_FILENAME = ".babelignore";
 
 function exists(filename) {
-  const cached = existsCache[filename];
-  if (cached == null) {
-    return existsCache[filename] = fs.existsSync(filename);
-  } else {
-    return cached;
+  if (existsCache[filename] == null) {
+    try {
+      existsCache[filename] = fs.existsSync(filename);
+    } catch (e) {
+      if (typeof fs.existsSync !== "function") {
+        throw new Error(
+          "You are using babel in an environment that does not include filesystem access. " +
+          "Ensure to disable .babelrc lookup by setting `babelrc: false` in your config."
```

Comment:
Hmm I thought it might be better to do it in the catch as in 99,9% of the cases the check is useless as babel i usually used in node, but probably deopt of v8 because of try catch hurts more.